### PR TITLE
Add experimental font size attribute to Navigation block

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Replaces the current font size sidebar control with the new font size attribute on the Navigation block. 

## How has this been tested?

Tested in browser.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/navigation/edit.js [hunk 1]
```diff
@@ -64,15 +59,13 @@ function Navigation( {
 				{
 					backgroundColor: true,
 					textColor: true,
-					fontSize: fontSize.size,
```
Comment (tellthemachines) [inline packages/block-library/src/navigation/edit.js:67]: With this new setup, how can we pass the font size value to the color panel contrast checker? I noticed that `ColorEdit ` and `ColorPanel` components don't explicitly deal with font size, and it's not being passed to the contrast checker in e.g. the Paragraph and the Heading block color panels.

File: packages/block-library/src/navigation/block.json [hunk 2]
```diff
@@ -50,13 +50,11 @@
 		"showSubmenuIcon": "showSubmenuIcon"
 	},
 	"supports": {
```
Comment (jorgefilipecosta) [inline packages/block-library/src/navigation/block.json:52]: Should we remove the fontSize attribute from block.json as we did for paragraph?

**Review by jorgefilipecosta — APPROVED** [review]
LGTM 👍

File: packages/block-library/src/navigation/edit.js (hunk 1 above)
Comment (tellthemachines) [reply packages/block-library/src/navigation/edit.js:67]: Cool, is this mentioned in any existing issue or shall I create a new one?

Comment (jorgefilipecosta) [reply packages/block-library/src/navigation/edit.js:67]: I am not aware of an issue for it so I guess we should create a new one.

Comment (tellthemachines) [reply packages/block-library/src/navigation/edit.js:67]: Done: #24607

**Pull request merged**